extensions/wikia/WallNotifications/tests/WallNotificationsControllerTest.php

```diff
@@ -29,7 +31,59 @@ public function testIndex( $isUserLoggedIn, $isUserAllowed, $wgAtCreateNewWikiPa
 		$this->mockGlobalVariable( 'wgUser', $userMock );
 
 		$resp = $this->app->sendRequest( 'WallNotificationsController', 'Index', [ 'format' => 'json' ] );
-		$this->assertEquals( $hasPrehide, is_bool($resp->getVal('prehide')) );
+		$this->assertEquals( $hasPrehide, is_bool( $resp->getVal( 'prehide' ) ) );
+	}
+
+
+	/**
+	 * @param $expectedMessageId
+	 * @param $expectedMessageTemplate
+	 * @param $expectedMessage
+	 * @param $username
+	 * @param $notifyGrouped
+	 * @dataProvider notificationDataProvider
+	 */
+	public function testNotify( $expectedMessageId, $expectedMessageTemplate, $expectedMessage, $username, $notifyGrouped ) {
+		//given
+		$userMock = $this->getMock( 'User', [ 'getName' ], [ ], '', false );
+
+		$userMock->expects( $this->any() )
+			->method( 'getName' )
+			->will( $this->returnValue( $username ) );
+		$userMock->mFrom = 'session';
+		$this->mockGlobalVariable( 'wgUser', $userMock );
+		$params =
+			[
+				'parent_id' => 1245,
+				'notify' => [ 'grouped' => self::getNotifyGroupMock( $notifyGrouped ) ]
+			];
+
+		$this->mockMessage( $expectedMessageId, $expectedMessageTemplate );
+
+		//when
+		$resp = $this->app->sendRequest( 'WallNotificationsController', 'Notification', $params );
+		var_dump( $resp->getVal( 'debug' ) );
```

Comment:
What this debug tells?
if we want to see it in the output maybe some caption would be welcomed here.
BTW, is it really a way we solve adding debug info to unit tests output?